packages/esm-patient-visits-report-app/src/hooks/useOPDCategories.tsx

```diff
@@ -37,65 +38,177 @@ export function useOPDCategories(initialCategory="outPatientClients") {
 		const [totalOpdVisits, setTotalOpdVisits] = useState(0);
 		const [totalOpdRevisits, setTotalOpdRevisits] = useState(0);
 
-    const getOPDVisits = async ({page, size}) => {
-      try {
-        setLoading(true);
-
-		const formatDateForReq = (dateString) => {
-			const date = new Date(dateString);
-      		const year = date.getFullYear();
-      		const month = String(date.getMonth() + 1).padStart(2, '0');
-      		const day = String(date.getDate()).padStart(2, '0');
-      		return `${day}-${month}-${year}`;
-		} 
-
-        let startString = formatDateForReq(dateRange.start);
-        let endString = formatDateForReq(dateRange.end);
-
-        const url = `/ws/rest/v1/ehospital/${category}?startDate=${startString}&endDate=${endString}&page=${page}&size=${size}`;
-        const {data} = await openmrsFetch(url);
-				const cardUrl = `/ws/rest/v1/ehospital/outPatientClients?startDate=${startString}&endDate=${endString}&page=${page}&size=${size}`;
-				const {data: cardData} = await openmrsFetch(cardUrl);
-
-        setData([])
-        if (data.results.length > 0) {
-					setData(prev => [...prev, ...data.results.map(result => ({
+    // const getOPDVisits = async ({page, size}) => {
+    //   try {
```

Comment:
@Samstar10 please clean up this section.
Or is there something that you are still working on this section?
cc: @PiusKariuki

Reply:
Yes @Michaelndula , there is something I am still working and using that block of code as a reference.